Yoshi/Yoshi/Menus/YoshiTableViewMenu/YoshiTableViewMenuCellDataSource.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  YoshiTableViewMenuCellDataSource.swift
+//  Yoshi
+//
+//  Created by Kanglei Fang on 24/02/2017.
+//  Copyright © 2017 Prolific Interactive. All rights reserved.
+//
+
+internal struct YoshiTableViewMenuCellDataSource: YoshiResuableCellDataSource {
+
+    private let title: String
+
+    private let subtitle: String?
+
+    init(title: String, subtitle: String?) {
+        self.title = title
+        self.subtitle = subtitle
+    }
+
+    func cellFor(tableView: UITableView) -> UITableViewCell {
```

Comment:
I feel the code of this function is duplicated except the reuseIdentifier, maybe you could move that to the extension and only have a reseuseIdentifer property define in the protocol and implemented in your struct